Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -6991,6 +6991,25 @@ static bool consumeShapeBorderRadius(CSSParserTokenRange& args, const CSSParserC
     return true;
 }
 
+static RefPtr<CSSRectShapeValue> consumeBasicShapeRect(CSSParserTokenRange& args, const CSSParserContext& context)
+{
+    std::array<RefPtr<CSSValue>, 4> offsets;
+    for (auto& offset : offsets) {
+        if (args.peek().id() == CSSValueAuto)
+            offset = consumeIdent<CSSValueAuto>(args);
+        else
+            offset = consumeLengthOrPercent(args, context.mode, ValueRange::All);
```

Comment:
Maybe make `consumeAutoOrLengthOrPercent` accessible outside CSSPropertyParserHelpers.cpp and give it arguments you could use here?